seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py

```diff
@@ -0,0 +1,225 @@
+        [
+            # cc_aligned_sent_source
+            SEACrowdConfig(
+                name=f"{_DATASETNAME}_source",
+                version=SOURCE_VERSION,
+                description=f"{_DATASETNAME} source schema (all)",
+                schema="source",
+                subset_id=_DATASETNAME,
+            ),
+            # cc_aligned_sent_t2t
+            SEACrowdConfig(
+                name=f"{_DATASETNAME}_seacrowd_{SEACROWD_SCHEMA_NAME}",
+                version=SEACROWD_VERSION,
+                description=f"{_DATASETNAME} SEACrowd schema (all)",
+                schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+                subset_id=_DATASETNAME,
+            ),
+        ]
+    )
+
+    # Choose first language as default
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "Source_Sentence": datasets.Value("string"),
+                    "Target_Sentence": datasets.Value("string"),
+                    "LASER_similarity": datasets.Value("float64"),
+                }
+            )
+
+        if self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            features = schemas.text_to_text.features
+
+        return datasets.DatasetInfo(description=_DESCRIPTION, features=features, homepage=_HOMEPAGE, license=_LICENSE, citation=_CITATION)
+
+    def _split_generators(self, dl_manager: DownloadManager) -> List[datasets.SplitGenerator]:
+        """Return SplitGenerators."""
+        source_files = []
+        source_languages = []
+        target_languages = []
+
+        # Define some functions for parsing config and URL names
+        def _split_at_n(text: str, n: int) -> Tuple[str, str]:
+            return ("_".join(text.split("_")[:n]), "_".join(text.split("_")[n:]))
+
+        def _get_source_target_langs(url: str) -> Tuple[str, str]:
+            return Path(url).stem.split(".")[0].split("-")
+
+        # Get language from config name
+        _, name = _split_at_n(self.config.name, 3)
+        lang, _ = name.split("_", 1)
+
+        if lang in _LANGUAGES:
+            # Load data per language
+            url = _URLS[lang]
+            source_lang, target_lang = _get_source_target_langs(url)
+
+            source_files.append(dl_manager.download_and_extract(_URLS[lang]))
+            source_languages.append(source_lang)
+            target_languages.append(target_lang)
+        else:
+            # Load examples for all languages at once
+            for lang in _LANGUAGES:
+                url = _URLS[lang]
+                source_lang, target_lang = _get_source_target_langs(url)
+
+                source_files.append(dl_manager.download_and_extract(_URLS[lang]))
+                source_languages.append(source_lang)
+                target_languages.append(target_lang)
```

Comment:
This also optional

Reply:
Updated in the latest commit. This simplified the code a lot! Thanks :) https://github.com/SEACrowd/seacrowd-datahub/commit/7b89f0cc5282faa1c7a95303ebe5dcfb652a741f